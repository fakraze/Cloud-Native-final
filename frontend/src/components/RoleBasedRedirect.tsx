import React from 'react';
import { Navigate } from 'react-router-dom';
import { useAuthStore } from '../store/authStore';

export const RoleBasedRedirect: React.FC = () => {
  const { user } = useAuthStore();

  if (!user) {
    return <Navigate to="/login" replace />;
  }

  // Redirect based on user role
  if (user.role === 'admin') {
    return <Navigate to="/admin" replace />;
  } else {
    return <Navigate to="/restaurant" replace />;
  }
};
