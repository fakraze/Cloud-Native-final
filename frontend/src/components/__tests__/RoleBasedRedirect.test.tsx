import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render } from '../../test/test-utils';
import { RoleBasedRedirect } from '../RoleBasedRedirect';

const mockAuthStore = vi.hoisted(() => ({
  useAuthStore: vi.fn()
}));

vi.mock('../../store/authStore', () => mockAuthStore);

describe('RoleBasedRedirect Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should redirect to login when no user', () => {
    mockAuthStore.useAuthStore.mockReturnValue({
      user: null
    });

    const { container } = render(<RoleBasedRedirect />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should redirect admin to admin dashboard', () => {
    mockAuthStore.useAuthStore.mockReturnValue({
      user: {
        id: '1',
        username: 'admin',
        role: 'admin'
      }
    });

    const { container } = render(<RoleBasedRedirect />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should redirect staff to staff dashboard', () => {
    mockAuthStore.useAuthStore.mockReturnValue({
      user: {
        id: '2',
        username: 'staff',
        role: 'staff'
      }
    });

    const { container } = render(<RoleBasedRedirect />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should redirect customer to restaurants', () => {
    mockAuthStore.useAuthStore.mockReturnValue({
      user: {
        id: '3',
        username: 'customer',
        role: 'customer'
      }
    });

    const { container } = render(<RoleBasedRedirect />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
