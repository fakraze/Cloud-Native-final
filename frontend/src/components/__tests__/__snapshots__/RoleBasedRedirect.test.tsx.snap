// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoleBasedRedirect Component > should redirect admin to admin dashboard 1`] = `null`;

exports[`RoleBasedRedirect Component > should redirect customer to restaurants 1`] = `null`;

exports[`RoleBasedRedirect Component > should redirect staff to staff dashboard 1`] = `null`;

exports[`RoleBasedRedirect Component > should redirect to login when no user 1`] = `null`;
